### PR TITLE
🌱 Update hcloudmachine_types.go

### PR DESCRIPTION
**Pull request by Sayanta66**
**What this PR does / why we need it**:

Added descriptions

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squash commits
- [x] include documentation
- [ ] add unit tests

### CODE REVIEW

File: api/v1beta1/hcloudmachine_types.go [hunk 1]
```diff
@@ -48,6 +48,7 @@ type HCloudMachineSpec struct {
 	// +optional
 	SSHKeys []SSHKey `json:"sshKeys,omitempty"`
 
+	// The placement group of the machine in HCloud API must be referencing an existing placement group
```
Comment (janiskemper) [inline api/v1beta1/hcloudmachine_types.go:51]:
```suggestion
	// The placement group of the machine in HCloud API must be referencing an existing placement group.
```

**Review by janiskemper — CHANGES_REQUESTED** [review]

**Comment by janiskemper** [comment]
thanks, you can squash

**Comment by Sayanta66** [comment]
Updating a few more descriptions in line with our reference docs. Then will squash all the commits together.

File: api/v1beta1/hcloudmachine_types.go [hunk 2]
```diff
@@ -36,23 +36,25 @@ type HCloudMachineSpec struct {
...
 	// +kubebuilder:validation:Enum=cpx11;cx21;cpx21;cx31;cpx31;cx41;cpx41;cx51;cpx51;ccx11;ccx12;ccx13;ccx21;ccx22;ccx23;ccx31;ccx32;ccx33;ccx41;ccx42;ccx43;ccx51;ccx52;ccx53;ccx62;ccx63;cax11;cax21;cax31;cax41
 	Type HCloudMachineType `json:"type"`
 
 	// ImageName is the reference to the Machine Image from which to create the machine instance.
+	// ImageName can reference an image uploaded to Hetzner API in two ways: either directly as the name of an image or as the label of an image.
 	// +kubebuilder:validation:MinLength=1
 	ImageName string `json:"imageName"`
 
-	// define Machine specific SSH keys, overrides cluster wide SSH keys
+	// Define Machine specific SSH keys, and override cluster-wide SSH keys.
```
Comment (janiskemper) [inline api/v1beta1/hcloudmachine_types.go:48]: Start with "SSHKeys ..."

**Review by janiskemper — CHANGES_REQUESTED** [review]

File: api/v1beta1/hcloudmachine_types.go [hunk 3]
```diff
@@ -36,22 +36,25 @@ type HCloudMachineSpec struct {
...
+	// ImageName can reference an image uploaded to Hetzner API in two ways: either directly as the name of an image or as the label of an image.
 	// +kubebuilder:validation:MinLength=1
 	ImageName string `json:"imageName"`
 
-	// define Machine specific SSH keys, overrides cluster wide SSH keys
+	// SSHKeys define Machine specific SSH keys, and override cluster-wide SSH keys.
 	// +optional
 	SSHKeys []SSHKey `json:"sshKeys,omitempty"`
 
+	// The placement group of the machine in HCloud API must be referencing an existing placement group.
```
Comment (janiskemper) [inline api/v1beta1/hcloudmachine_types.go:52]: Start with PlacementGroupName

File: api/v1beta1/hcloudmachine_types.go [hunk 4]
```diff
@@ -36,22 +36,25 @@ type HCloudMachineSpec struct {
...
 	// +optional
 	PlacementGroupName *string `json:"placementGroupName,omitempty"`
 
-	// PublicNetwork specifies information for public networks
+	// PublicNetwork specifies information for public networks.
```
Comment (janiskemper) [inline api/v1beta1/hcloudmachine_types.go:56]: this line and the following line don't fit together

**Review by janiskemper — CHANGES_REQUESTED** [review]

File: api/v1beta1/hcloudmachine_types.go [hunk 5]
```diff
@@ -36,22 +36,25 @@ type HCloudMachineSpec struct {
 	// +optional
 	ProviderID *string `json:"providerID,omitempty"`
 
-	// Type is the HCloud Machine Type for this machine.
+	// Type is the HCloud Machine Type for this machine. It defines the desired server type of server in Hetzner's Cloud API. Example: cpx11.
 	// +kubebuilder:validation:Enum=cpx11;cx21;cpx21;cx31;cpx31;cx41;cpx41;cx51;cpx51;ccx11;ccx12;ccx13;ccx21;ccx22;ccx23;ccx31;ccx32;ccx33;ccx41;ccx42;ccx43;ccx51;ccx52;ccx53;ccx62;ccx63;cax11;cax21;cax31;cax41
 	Type HCloudMachineType `json:"type"`
 
 	// ImageName is the reference to the Machine Image from which to create the machine instance.
+	// ImageName can reference an image uploaded to Hetzner API in two ways: either directly as the name of an image or as the label of an image.
```
Comment (janiskemper) [inline api/v1beta1/hcloudmachine_types.go:44]: you don't need to start twice with "ImageName". Start with "it" or something

File: api/v1beta1/hcloudmachine_types.go [hunk 6]
```diff
@@ -36,22 +36,25 @@ type HCloudMachineSpec struct {
...
 	// +kubebuilder:validation:MinLength=1
 	ImageName string `json:"imageName"`
 
-	// define Machine specific SSH keys, overrides cluster wide SSH keys
+	// SSHKeys define Machine-specific SSH keys and override cluster-wide SSH keys.
```
Comment (janiskemper) [inline api/v1beta1/hcloudmachine_types.go:48]: Machine -> machine

**Review by janiskemper — CHANGES_REQUESTED** [review]

**Review by janiskemper — CHANGES_REQUESTED** [review]
there is an empty introduction.md file in this commit! You should remove it

**Comment by Sayanta66** [comment]
> there is an empty introduction.md file in this commit! You should remove it

Done

**Review by janiskemper — APPROVED** [review]

**Pull request merged**